src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Symbolic/SymbolicRegexBuilder.cs

```diff
@@ -43,6 +43,9 @@ internal sealed class SymbolicRegexBuilder<TElement> where TElement : notnull
         // states that have been created
         internal HashSet<DfaMatchingState<TElement>> _stateCache = new();
 
+        // capturing states that have been created
+        internal HashSet<DfaMatchingState<TElement>> _capturingStateCache = new();
```

Comment:
Nit: it'd be nice if these were readonly; we can clean that up subsequently, though.